common/djangoapps/student/views.py

```diff
@@ -2486,18 +2486,22 @@ def validate_password(user, password):
         password: the user's proposed new password.
 
     Returns:
-        is_valid_password: a boolean indicating if the new password
-            passes the validation.
+        is_security_violated: a boolean indicating if the new password
+            passes the security policy validation.
         err_msg: an error message if there's a violation of one of the password
             checks. Otherwise, `None`.
     """
     err_msg = None
 
-    if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
-        try:
+    try:
+        if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
             validate_password_strength(password)
-        except ValidationError as err:
-            err_msg = _('Password: ') + '; '.join(err.messages)
+        else:
+            validate_password_length(password)
+
+    except ValidationError as err:
```

Comment:
Did you know that ValidationError is raised by both the `validate_password_strength()` and `validate_password_length()` functions.
You have to distinguish the validationError based on it's `code` property.
The way you are handling validationError here, you are also catching the validationError raised by `validate_password_strength()` function. Your comment below is invalid.
Why do we need to distinguish the password invalidity by length or by strength? Shouldn't we treat them the same and have the same messages? From what I can tell, the `validate_password_strength()` and `validate_password_length()` functions all give pretty good error messages in the validationError exception